lms/services/blackboard_api/client.py

```diff
@@ -113,4 +113,56 @@ def group_set_groups(self, course_id, group_set_id):
             "GET",
             f"/learn/api/public/v2/courses/uuid:{course_id}/groups/sets/{group_set_id}/groups",
         )
-        return BlackboardListGroupSetGroups(response).parse()
+        return BlackboardListGroups(response).parse()
+
+    def course_groups(
+        self, course_id, group_set_id=None, current_student_own_groups_only=True
+    ):
+        """
+        Return the groups in a course.
+
+        :param course_id: ID of the course
+        :param group_set_id: Only return groups that belong to this group set
+        :param current_student_own_groups_only: Only return groups the current user (a student) belong to.
+        """
+        response = self._api.request(
+            "GET",
+            f"/learn/api/public/v2/courses/uuid:{course_id}/groups",
+        )
+        groups = BlackboardListGroups(response).parse()
+
+        if group_set_id:
+            groups = [group for group in groups if group["groupSetId"] == group_set_id]
+
+        if not current_student_own_groups_only:
+            return groups
+
+        instructor_only_groups = []
+        self_enrollment_groups = []
+        self_enrollment_check_urls = []
+
+        for group in groups:
```

Comment:
We need to split the groups between InstructorOnly/SelfEnrollment and generate the URLs so one pass in a for loop ended up being nicer than multiple list comprehensions or something else.